### PR TITLE
US132357 - Add more unit tests and demos

### PR DESCRIPTION
**Pull request by semibran**
Lit flagging is broken on Safari 14 until we figure out the FRA stuff. It's also using the `rubrics-lit` flag which isn't necessarily a bad thing per se, we just need to determine whether we actually want one flag per conversion project or not and rename the flag to something more accurate if necessary. Anyways I'll get that all sorted out in future PRs, this ticket was just supposed to be for unit tests 😰 

https://rally1.rallydev.com/#/57732444928d/custom/373260458992?detail=%2Fuserstory%2F610504275289

### CODE REVIEW

File: test/d2l-user-alignment-list.test.js [hunk 1]
```diff
@@ -1,10 +1,57 @@
...
 	});
+
+	it('resolves hypermedia entity', async() => {
+		const element = await fixtureWaitForChangeEvent(html`
+			<d2l-user-alignment-list></d2l-user-alignment-list>
+		`, 'static-data/alignment-collections/e1a717aa-355b-4623-a9b6-88aa110df4de.json');
+		expect(element.entity).to.be.instanceof(UserAlignmentListEntity);
+	});
+
+	describe('test responsiveness', () => {
```
Comment (semibran) [inline test/d2l-user-alignment-list.test.js:21]: This whole suite sucks. I think I was gonna add a whole bunch of stuff around the LOA scales which is why I slogged through shipwreck raws to get all the mock data around them but then once I got it working I just blanked out really hard. At least the demo works fine..... we can probably just wipe this section if/once we ever do vdiffs in this repo

File: test/helpers/fixture-load-custom-data.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+		}));
+};
+
+window.D2L.PolymerBehaviors.FetchSirenEntityBehavior._makeRequest = (request) => {
+	const href = request.url && request.url.href || request.url;
+	if (href === globals.href) {
+		return Promise.resolve(SirenParse(globals.data));
+	}
+};
+window.D2L.Siren.WhitelistBehavior._testMode(true);
```
Comment (imcdonald-d2l) [inline test/helpers/fixture-load-custom-data.js:38]: What is the purpose of these modification on `window.{}` ? I'm not seeing them being used in the exported `loadCustomData()` function and can't seem to tell where they are being used.

Comment (semibran) [reply test/helpers/fixture-load-custom-data.js:38]: Yeah so this is all stuff that was previously being used in [demo/request-mock.js](https://github.com/Brightspace/d2l-activity-alignments/blob/2.x/demo/request-mock.js) and [test/d2l-outcome.html](https://github.com/Brightspace/d2l-activity-alignments/blob/2.x/test/d2l-outcome.html)...... it's basically a hack that lets us inline custom HM entity mock data by overriding d2lfetch. If anyone knows of a not-as-gross way to do all of this stuff I'm all ears.

File: test/d2l-user-alignment-list.test.js (hunk 1 above)
Comment (imcdonald-d2l) [reply test/d2l-user-alignment-list.test.js:21]: Yeah, makes sense to wipe this section if vdiffs ever happen

**Review by imcdonald-d2l — APPROVED** [review]

**Pull request merged**